File: cpp/001-010/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        unordered_map<int, int> m;
        vector<int> result;
        for (int i = 0;i < numbers.size();i++) {
            int remain = target - numbers[i];
            if (m.count(remain) > 0) {
                result.push_back(m[remain]);
                result.push_back(i);
                return result;
            }
            m[numbers[i]] = i;
        }
        return result;
    }
};

File: golang/001-010/Two Sum.go
func twoSum(nums []int, target int) []int {
    mymap := make(map[int]int)
    for i := 0; i < len(nums); i++ {
        j, ok := mymap[target-nums[i]]
        if ok {
            result := []int{j, i}
            return result
        }
        mymap[nums[i]] = i
    }
    result := []int{-1, -1}
    return result
}
